tensorboard/summary/BUILD

```diff
@@ -1,20 +1,81 @@
 # Description:
 # Summary API for TensorBoard.
 
+load("//tensorboard/defs:py_repl.bzl", "py_repl")
+
 package(default_visibility = ["//tensorboard:internal"])
 
 licenses(["notice"])
 
+# Standard TensorBoard summary writer API build target, which
+# contains all optional dependencies. For a slimmer target see
+# `:summary_minimal`.
 py_library(
     name = "summary",
+    deps = [
+        ":summary_minimal",
+    ],
+)
+
+# Minimal dependency for the TensorBoard summary writer APIs.
+# Does not contain any optional deps, so some APIs may throw
+# exceptions when called if not all their deps are satisfied.
+py_library(
+    name = "summary_minimal",
+    srcs = ["__init__.py"],
+    srcs_version = "PY3",
+    deps = [
+        ":writer",
+    ],
+)
+
+py_library(
+    name = "writer",
     srcs = [
-        "__init__.py",
+        "_writer.py",
     ],
     srcs_version = "PY3",
-    visibility = ["//tensorboard:internal"],
     deps = [
-        ":summary_v1",
-        ":summary_v2",
+        ":output",
+        "//tensorboard:expect_numpy_installed",
+    ],
+)
+
+py_test(
+    name = "writer_test",
+    srcs = ["writer_test.py"],
```

Comment:
Ok, I've updated them all to have a leading underscore.  It's slightly awkward-feeling in practice IMO, since the underscore starts getting kind of viral (leaks into BUILD targets, for the test util also into imports in the other files, etc.).  But I agree it is more consistent overall.